src/stream/src/executor/source.rs

```diff
@@ -41,8 +41,12 @@ use crate::executor::{Executor, ExecutorBuilder, Message, PkIndices, PkIndicesRe
 use crate::task::{ExecutorParams, LocalStreamManagerCore};
 
 struct SourceReader {
+    /// the future that builds stream_reader. It is required because source should not establish
+    /// connections to the upstream before `next` is called
```

Comment:
We can avoid this when we migrate `SourceExecutor` to executor v2. 😁

Reply:
next time for sure 👍

The migration will happen after making source running!